pkg/controller/connectivitycheck/connectivity_check_controller.go

```diff
@@ -0,0 +1,277 @@
+package connectivitycheck
+
+import (
+	"context"
+	"net"
+	"net/url"
+	"os"
+	"strconv"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/api/operatorcontrolplane/v1alpha1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	configv1listers "github.com/openshift/client-go/config/listers/config/v1"
+	operatorcontrolplaneclient "github.com/openshift/client-go/operatorcontrolplane/clientset/versioned"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/connectivitycheckcontroller"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/genericoperatorclient"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apiextensionsinformers "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+type OVNKubernetesConnectivityCheckController interface {
+	connectivitycheckcontroller.ConnectivityCheckController
+}
+
+func NewOVNKubernetesConnectivityCheckController(
+	kubeClient kubernetes.Interface,
+	operatorClient v1helpers.OperatorClient,
+	operatorcontrolplaneClient *operatorcontrolplaneclient.Clientset,
+	apiextensionsClient *apiextensionsclient.Clientset,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	configInformers configinformers.SharedInformerFactory,
+	apiextensionsInformers apiextensionsinformers.SharedInformerFactory,
+	recorder events.Recorder,
+) OVNKubernetesConnectivityCheckController {
+	c := ovnKubernetesConnectivityCheckController{
+		ConnectivityCheckController: connectivitycheckcontroller.NewConnectivityCheckController(
+			"openshift-ovn-kubernetes",
+			operatorClient,
+			operatorcontrolplaneClient,
+			apiextensionsClient,
+			apiextensionsInformers,
+			[]factory.Informer{
+				operatorClient.Informer(),
+				kubeInformersForNamespaces.InformersFor("openshift-ovn-kubernetes").Core().V1().Pods().Informer(),
```

Comment:
Ideally we should not have hardcoded references to ovn-kubernetes anywhere in the code except `pkg/network/ovn_kubernetes.go`. The preferred way to do something like this is to add annotations to objects in `bindata/network/ovn-kubernetes/` and then act generically on the annotations here. (`pkg/controller/operconfig` can notice when it creates/reconciles an object with annotations and then inform this controller, if that would help)

(Except it sounds like probably this isn't going to involve the ovn-kubernetes namespace at all anyways)